src/topology/algebra/continuous_monoid_hom.lean

```diff
@@ -0,0 +1,134 @@
+/-
+Copyright (c) 2022 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+import topology.algebra.group
+import topology.continuous_function.basic
+
+/-!
+
+# Continuous Monoid Homs
+
+This file defines the space of continuous homomorphisms between two topological groups.
+
+## Main definitions
+
+* `continuous_monoid_hom A B`: The continuous homomorphisms `A →* B`.
+
+-/
+
+variables (A B C D E : Type*)
+  [monoid A] [monoid B] [monoid C] [monoid D] [comm_group E]
+  [topological_space A] [topological_space B] [topological_space C] [topological_space D]
+  [topological_space E] [topological_group E]
+
+set_option old_structure_cmd true
+
+/-- Continuous homomorphisms between two topological groups. -/
+structure continuous_monoid_hom extends A →* B, continuous_map A B
+
+/-- Continuous homomorphisms between two topological groups. -/
+structure continuous_add_monoid_hom (A B : Type*) [add_monoid A] [add_monoid B]
+  [topological_space A] [topological_space B] extends A →+ B, continuous_map A B
+
+attribute [to_additive] continuous_monoid_hom
+attribute [to_additive] continuous_monoid_hom.to_monoid_hom
+attribute [nolint doc_blame] continuous_monoid_hom.to_monoid_hom
+attribute [nolint doc_blame] continuous_monoid_hom.to_continuous_map
+attribute [nolint doc_blame] continuous_add_monoid_hom.to_add_monoid_hom
+attribute [nolint doc_blame] continuous_add_monoid_hom.to_continuous_map
+
+namespace continuous_monoid_hom
+
+variables {A B C D E}
+
+@[to_additive] instance : has_coe_to_fun (continuous_monoid_hom A B) (λ _, A → B) :=
+⟨continuous_monoid_hom.to_fun⟩
+
+@[to_additive] lemma ext {f g : continuous_monoid_hom A B} (h : ∀ x, f x = g x) : f = g :=
+by cases f; cases g; congr; exact funext h
+
+/-- Construct a `continuous_monoid_hom` from a `continuous` `monoid_hom`. -/
+@[to_additive "Construct a `continuous_add_monoid_hom` from a `continuous` `add_monoid_hom`."]
+def mk' (f : A →* B) (hf : continuous f) : continuous_monoid_hom A B := { .. f }
```

Comment:
Please provide simp-lemmas for what happens when you apply `mk' f hf` to some `a : A`. Idem for all the other defs below. If you setup simps projections, this might be as easy as adding `@[simps]` to all the defs.

Search mathlib for `initialize_simps_projections`.

Note that `simps` should come *after* `to_additive` when both are used.